crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -1,4 +1,80 @@
+// straight away in the incoming dht ops workflow. see https://github.com/holochain/holochain/issues/3724
+
+//! # Op validation
+//!
+//! For each op the [corresponding app validation function](https://docs.rs/hdi/latest/hdi/#data-validation)
+//! is executed. Entries and links, that the ops are derived from, have been
+//! written with a particular integrity zome's entry and link types. Thus for
+//! op validation, the validation function of the same integrity zome must be
+//! used. Ops that do not relate to a specific entry or link like [`DhtOp::RegisterAgentActivity`]
+//! or non-app entries like [`EntryType::CapGrant`] are validated with all
+//! validation functions of the DNA's integrity zomes.
+//!
+//! Having established the relevant integrity zomes for validating an op, each
+//! zome's validation callback is invoked.
+//!
+//! # Outcome
+//!
+//! An op can be valid or invalid, which is considered "validated", or it could not be
+//! validated because it is missing required dependencies such as actions,
+//! entries, links or agent activity that the validation function is referencing. If
+//! all ops were validated, the workflow completes with no further action. If
+//! however some ops could not be validated, the workflow will trigger itself
+//! again after a delay, while missing dependencies are being fetched in the
+//! background.
+//!
+//! # Missing dependencies
+//!
+//! Finding the zomes to invoke for validation oftentimes involves fetching a
+//! referenced original action, like in the case of updates and deletes. Further
+//! the validation function may require actions, entries or agent activity
+// (segments of an agent's source chain) that currently are not stored in the
+//! local databases. These are dependencies of the op. If they are missing
+//! locally, a network get request will be sent in the background. The op
+//! validation outcome will be [`Outcome::AwaitingDeps]. Validation of
+//! remaining ops will carry on, as the network request's response is not
+//! awaited within the op validation loop. Instead the whole workflow triggers
+//! itself again after a delay.
+//!
+//! # Workflow re-triggering
+//!
+//! Ops that are known to have missing dependencies are omitted from a workflow
+//! run, because they cannot be validated without them. Once their missing
+//! dependencies have all been fetched, these ops will be validated the next
+//! time the workflow runs.
+//!
+//! Awaiting missing ops re-triggers the validation workflow for a fixed period
+//! of time. After this period has elapsed without new missing ops having
+//! arrived, workflow re-triggering ends.
+//!
+//! # Integration workflow
+//!
+//! If any ops have been validated (outcome valid or invalid), the [`integrate_dht_ops_workflow](crate::core::workflow::integrate_dht_ops_workflow)
+//! is triggered, that completes integration of ops after successful validation.
+// This seems to mainly affect ops with system dependencies, as ops without such
```

Comment:
As with the previous non-documentation comment, does this refer to a statement above or to code below? And it sounds like it's saying ops without system deps are integrated immediately rather than being farmed out to the integration workflow?

Reply:
Previous again, and yes, they're integrated immediately.